core/ResManager/plResManager.h

```diff
@@ -486,6 +487,9 @@ class HSPLASMA_EXPORT plResManager
 
     /** Unlock access to the ResManager in multithreaded apps */
     void unlock() { fResMgrMutex.unlock(); }
+
+    /** Sets a flag to preserve existing Object IDs on read/write. */
+    void preserveObjectIDs() { preserveIDs = true; }
```

Comment:
I think I'd rather see this as an optional parameter in the constructor...  When it's written here, it's a one-way flag anyway (there's no way to unset it), and it would still need to be done before any reads or writes were started.